statsmodels/genmod/generalized_linear_model.py

```diff
@@ -700,18 +713,82 @@ def estimate_tweedie_power(self, mu, method='brentq', low=1.01, high=5.):
         power : float
             The estimated shape or power
         """
+        if exposure is not None and not isinstance(self.family.link,
+                                                   families.links.Log):
+            raise ValueError("exposure can only be used with the log link function")
+
+        # Use fit exposure if appropriate
+        if exposure is None and hasattr(self, 'exposure'):
+            # Already logged
+            exposure = np.exp(self.exposure)
+        elif exposure is None:
+            exposure = np.ones(len(self.endog))
+
+        # Use fit freq_weights if appropriate
+        if freq_weights is None and hasattr(self, 'freq_weights'):
+            # Already logged
+            freq_weights = self.freq_weights
+        elif exposure is None:
+            freq_weights = np.ones(len(self.endog))
+
+        weights = freq_weights * exposure
```

Comment:
It's likely that what you have is correct, but it's not written in a way how it would follow from an explicit derivation of the formulas. 
The variance of uncorrected endog should be linear in exposure with exposure corrected mu in the power mu2**p.  var(y) = exposure * s2 * mu2**p   where s2 and mu2 are scale and mean at exposure=1, i.e. per unit of exposure.